Add Data File parser

0ddlyoko opened the pull request:
This PR will add the parser of data files

Minecraft data are taken from https://github.com/PrismarineJS/minecraft-data

For now, only blocks are imported. We'll import other data later

0ddlyoko committed:
[ADD] Add a parser to parse minecraft data files

This commit adds a parser to be allowed to import minecraft data files.

With that, we will be allowed to retrieve a minecraft id from a block state and vice versa for each different version of minecraft
For now, only data > 1.13 are imported

0ddlyoko commented:
Done & ready for a review
Only versions >= 13.0 works, we will handle versions < 13.0 later

hydrostaticcog commented on classes/utils/Utils.py:
```diff
@@ -49,23 +60,25 @@ def __gt__(self, other):
-    v1_16_0 = (1, 16, 0, 735)
-    v1_16_1 = (1, 16, 1, 736)
-    v1_16_2 = (1, 16, 2, 751)
+    v1_12_0 = (1, 12, 0, 335, '1.12')
```
loading all of the versions takes a while to load/its not really necessary.

hydrostaticcog commented on main.py:
```diff
@@ -328,6 +328,35 @@ def _launch_server():
+    from classes.utils.Parser import Parser
+    from classes.utils.Utils import Version
+    versions_string = [
+        '1.13',
```
we might want to hardcode this at some point...

hydrostaticcog reviewed (commented):
Looks good. One thing: this currently loads multiple versions of data when we only need/are testing 1.15.2. Also, the logging in main.py has hardcoded data logs. We should change this to only log what version of data that we are loading/what we need to load.

0ddlyoko commented:
> Looks good. One thing: this currently loads multiple versions of data when we only need/are testing 1.15.2. Also, the logging in main.py has hardcoded data logs. We should change this to only log what version of data that we are loading/what we need to load.

Thanks for the review, it's strange because I'm pretty sure I have updated the code.

I think I have updated the code and I haven't pushed it, fool than I am

hydrostaticcog commented:
Developer Left Project/PR left with conflicts

Geolykt commented:
I have looked at the PR again (given it's relative importance) and if the merge conflicts are naively resolved it still will not start up at all due to a circulatory import. I will look at what I can do to resolve that however
